### ISSUE TITLE
[Suggestion] option for non infinite water sources

### ISSUE THREAD

**Issue by Biviho** [jbredwards/Fluidlogged-API#32]
using this mod with for example better with mods will not respect it's feature for non infinite water sources.

If it's something you can be interested to provide (and it's not to difficult to achieve), i suggest something like some option in a config that will:

-drop flowing water when the fluidlogged block is broken
-revert the block into it's normal form if the fluid is not encased (example: one stair surrounded by blocks, after the block in front of it is removed)
-remove the fluid and revert the block into it's normal form when right clicked with an empty bucket, without filling it with the fluid
-add water when right clicked with a full bucket without removing the fluid from the bucket

basically, only the ability to activate the cosmetic part, without having the fluid mechanic.

let me know what you think of this

**Comment by jbredwards** [comment]
1: The issue with the Better With Mods mod sounds like it's on their side, report it to them. Correctly coded fluid blocks have a built-in true/false value that determines if the block can create infinite sources. If the mod doesn't update that value, or respect the value set by other mods, there will be problems.
2: This is a known issue that has a fix ready for the next update.
3: Correct me if I'm wrong, but as I understand this, you want a fluidlogged block to become un-fluidlogged when one side is exposed to air. I won't add this since it's inconsistent with the parent fluid's behavior.

In general, I want the fluidlogged blocks to be as consistent with the parent fluid block as possible, so I won't add a config for 'cosmetic only' fluidlogged blocks. This feature would also require a rewrite of most of the mod. Something like a rewrite would inconvenience any mod devs using this mod in their own stuff, and I wouldn't really want to do that unless there was some a core issue with the mod I don't currently know about or something. All in all, I won't add the 'cosmetic only' config because of the potential inconsistencies and other issues that would result from it. If you have a way of adding it that wouldn't result in this, feel free to leave a pull request and I'll take a look. :)
